storage-operator/pkg/salt/client.go

```diff
@@ -0,0 +1,253 @@
+package salt
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// A Salt API client.
+type Client struct {
+	address string       // Address of the Salt API server.
+	client  *http.Client // HTTP client to query Salt API.
+	creds   *Credential  // Salt API authentication credentials.
+	token   *authToken   // Salt API authentication token.
+	logger  logr.Logger  // Logger for the client's requests
+}
+
+// Create a new Salt API client.
+func NewClient(creds *Credential) *Client {
+	const SALT_API_PORT int = 4507 // As defined in master-99-metalk8s.conf
+
+	address := os.Getenv("METALK8S_SALT_MASTER_ADDRESS")
+	if address == "" {
+		address = "http://salt-master"
```

Comment:
by default minions looks for hostname `salt` , why we re looking for salt-master

Reply:
IIUC, we want to target the DNS entry for the pod, and when you're inside the `kube-system` namespace the pod is addressable through the `salt-master` hostname.

Reply:
Can't you rely on the `salt-api` port defined in the `salt-master` Service as well, instead of hardcoding it?

Reply:
We could, if there is an easy what to retrieve it.
It it's a bit "complicated" we can file a debt issue for it, (but if it's simple I'll add a commit ;))

Reply:
As seen with you, seems a bit involved, so I'll create a debt issue for that one.